File: wj-frontend/src/models/AiPlanRequest.ts
export class AiPlanRequest {
    constructor(
        public goal: string,
        public level: string,
        public daysPerWeek: string,
        public gender: string,
        public weight: string,
        public age: string,
        public height: string,
        public health: string,
        public intensity: string,
        public specialGoal: string
    ) {
    }
}

File: wj-backend/src/main/java/ovh/workoutjournal/requestmodels/AiPlanRequest.java
package ovh.workoutjournal.requestmodels;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class AiPlanRequest {

    @NotBlank
    private String goal;

    @NotBlank
    private String level;

    @NotBlank
    private String daysPerWeek;

    @NotBlank
    private String gender;

    @NotBlank
    private String weight;

    @NotBlank
    private String age;

    @NotBlank
    private String height;

    private String health;

    @NotBlank
    private String intensity;

    private String specialGoal;
}
